apps/remote_control/lib/lexical/remote_control/code_mod/extract_function.ex

```diff
@@ -0,0 +1,245 @@
+defmodule Lexical.RemoteControl.CodeMod.ExtractFunction do
+  alias Sourceror.Zipper, as: Z
+  @doc """
+  Return zipper containing AST with extracted function.
+  """
+  def extract_function(zipper, start_line, end_line, function_name)
+      when is_binary(function_name) do
+    extract_function(zipper, start_line, end_line, String.to_atom(function_name))
+  end
+
+  def extract_function(%Z{} = zipper, start_line, end_line, function_name)
+      when is_integer(start_line) and is_integer(end_line) and is_atom(function_name) do
+    {quoted_after_extract, acc} = extract_lines(zipper, start_line, end_line, function_name)
+
+    if Enum.empty?(acc.lines) do
+      {:error, :not_extractable}
+    else
+      new_function_zipper = Z.zip(new_function(function_name, [], acc.lines))
+      declared_vars = Enum.uniq(vars_declared(new_function_zipper))
+      used_vars = Enum.uniq(vars_used(new_function_zipper))
+
+      args = used_vars -- declared_vars
+      returns = declared_vars |> Enum.filter(&(&1 in acc.vars))
+
+      {zipper, extracted} =
+        add_returned_vars(Z.zip(quoted_after_extract), returns, function_name, args, acc.lines)
```

Comment:
another pipe-able function
```suggestion
       quoted_after_extract
       |> Z.zip()
       |> add_returned_vars(returns, function_name, args, acc.lines)
```